src/shared/utilities/workspaceUtils.ts

```diff
@@ -17,6 +17,15 @@ export function tryGetAbsolutePath(folder: vscode.WorkspaceFolder | undefined, r
     return path.resolve(folder?.uri ? folder.uri.fsPath + '/' : '', relPath)
 }
 
+export async function addWorkspaceFolder(folder: { uri: vscode.Uri; name?: string }): Promise<void> {
+    // No-op if the folder is already in the workspace.
+    if (vscode.workspace.getWorkspaceFolder(folder.uri)) {
+        return
+    }
+
+    await addFolderToWorkspace(folder)
```

Comment:
we have `addFolderToWorkspace` and `addWorkspaceFolder`. Two similar functions with no real distinction except the no-op behavior. We can eliminate the extra `addWorkspaceFolder` function, and just add an optional parameter to `addFolderToWorkspace`:

```typescript
/** ...
 * @param skipExisting  Do nothing if `folder` is already a workspace
*/
export async function addFolderToWorkspace(folder: { uri: vscode.Uri; name?: string }, skipExisting?: boolean): ... {
    // No-op if the folder is already in the workspace.
    if (!!skipExisting && vscode.workspace.getWorkspaceFolder(folder.uri)) {
        return
    }
}
```

I know this was old code, but... since we're already moving the function, it's a good time to just eliminate it.